cmd/mount.go

```diff
@@ -415,8 +427,27 @@ var mountCmd = &cobra.Command{
 
 				err = runPipeline(pipeline, ctx)
 				if err != nil {
+					errMsg, _ := ioutil.ReadFile(dmnCtx.LogFileName)
+					log.Err("libfuse exit with error message: %s", errMsg)
 					return err
 				}
+			} else {
+				// set signal handler
+				sigusr2 := make(chan os.Signal, 1)
+				signal.Notify(sigusr2, syscall.SIGUSR2)
```

Comment:
We shall register this before forking the child as it may happen that by the time code reaches here, child has already started and succeeded with sending a singla. As parent has not registered yet that signal will be lost and then parent will wait till child completes. You can test this scenario out by just adding a sleep of 1-2 seconds as first statement in the else condition.

Reply:
Nice catch!